### ISSUE TITLE
Attention mask cannot be passed into the forward function

### ISSUE THREAD

**Issue by vectominist** [jimbozhang/xares-llm-template#2]
According to the [xiaomi-research/xares-llm](https://github.com/xiaomi-research/xares-llm) repository, an `audio_attention_mask` can be passed into the forward function. E.g., in [example/dasheng/dasheng_base.py](https://github.com/xiaomi-research/xares-llm/blob/main/example/dasheng/dasheng_base.py)

```python
def forward(self, audio: torch.Tensor, audio_attention_mask=None) -> tuple[torch.Tensor, torch.Tensor | None]:
    assert isinstance(audio, torch.Tensor)
    if audio.ndim == 1:
        audio = audio.unsqueeze(0)
        features = self.processor(audio, return_tensors="pt")
        output = self.model(**features).hidden_states
    return output, None
```

However, the [template](https://github.com/jimbozhang/xares-llm-template/blob/main/template.py) here doesn't allow that

```python
def forward(self, audio: torch.Tensor):
    """
    Processes the input audio signal and returns the model's encoded output.
    Args:
        audio (torch.Tensor): A single-channel audio signal, represented as a PyTorch tensor with shape [B, T].
    Returns:
        torch.Tensor: The model's encoded output. The output tensor has shape [B, T', D], where T' can be different from the input T.
    """
    if audio.ndim == 1:
        audio = audio.unsqueeze(0)  # For robust, but NOT required to do this

    self.model.eval()
    with torch.inference_mode():
        encoded_audio = self.model(audio)

    return encoded_audio
```

Which template should we follow?

I'm asking this because whether an attention mask is provided changes the output of some transformer-based models with a batch of variable-length inputs, unless the batch size is always one.

**Comment by 907711588** [comment]
> Hey [@vectominist](https://github.com/vectominist) , Sorry for the late reply.
> 
> The first one is the correct one. By default we use an attention mask in order to avoid padding issues with llm training.
> 
> Kind regards, Heinrich

Could you update the Submission Rules or audio_encoder_checker.py? The current audio_encoder_checker.py requires the encoder to return a single tensor, but it should be updated to accommodate a (tensor, mask) format as well.
